Yoast/wordpress-seo
# P3-41 Create keyphrases table

## Context


*

## Summary


This PR can be summarized in the following changelog entry:

* Adds the SEMrush keyphrases table component

## Relevant technical choices:

- Introduces a new `AreaChart` component to generate SVG area charts.. This new component should be generic enough to be reused for any kind of area chart.
- Note: for now, the SEMrush response object is "mocked" and hardcoded as a default prop. I grabbed and example of the response data object from the real SEMrush API so hopefully the format is correct. There nay be the need for adjustments when the data will come from the real response.
- **Update:** after architects feedback, this PR also moves the `HelpLink` component from `js/src/components/contentAnalysis/HelpLink.js` to `js/src/components/HelpLink.js` so it's not too tied to the content analysis 

## Test instructions

This PR can be tested by following these steps:

Please test this together with https://github.com/Yoast/javascript/pull/799.

- build 
- edit a post, insert a keyphrase
- open the related keyphrases modal
- see the table within the modal
- check it matches the design 
- design for the table on Free: https://www.sketch.com/s/1ede3372-d1cc-4803-a4af-a06ca86aa0e3/a/9Qb7px
- on Premium, the table will have a 4th column with the buttons: I already added the code necessary to render the 4th column as implemented in https://yoast.atlassian.net/browse/P3-8 
- not sure there's a good way to test the Premium 4th column as P3-8 is still to be merged
- check the "Get more insights ar SEMrush" link at the bottom of the table
- the table should pass the main keyphrase as a parameter
- note: this link should also pass the country code (the SEMrush "database"), I added a note on https://yoast.atlassian.net/browse/P3-40 to cover this item
- ~the Help links within the table Headers are not implemented yet, pending final feedback from design/product~
- check the Help links as described in the comment below https://github.com/Yoast/wordpress-seo/pull/15673#issuecomment-661736319

To test with the correct styling:
* Make sure to check out the `P3-41-create-keyphrases-table` branch in the monorepo
* Run `yarn link-monorepo` in this branch
* Build and perform the test steps as mentioned above

## UI changes

## Documentation

## Quality assurance


Fixes [P3-41]
